src/main/java/org/generationcp/middleware/service/impl/dataset/DatasetServiceImpl.java

```diff
@@ -1018,6 +1023,14 @@ public void importDataset(final Integer datasetId, final Table<String, String, S
 							}
 
 						}
+						if (measurementVariable.getDataTypeId() == TermId.DATE_VARIABLE.getId()) {
+							try {
+								// In case the date is in yyyy-MM-dd format, try to parse it as number format yyyyMMdd
+								importedVariableValue = Util.convertDate(importedVariableValue, Util.FRONTEND_DATE_FORMAT, Util.DATE_AS_NUMBER_FORMAT);
+							} catch (ParseException e) {
+								// Do nothing
+							}
```

Comment:
Can we use `tryParseDate`?

Reply:
I just created a new method tryConvertDate to avoid the try-catch block here.

Reply:
I guess that's fine. But another way could be to just ignore the old format and try to parse the new one.
And for isValid, another option would be to overload it with `isValid(date, format)`

@abatac